File: recovered/src/com/lumiyaviewer/lumiya/render/GPUDetection.java
package com.lumiyaviewer.lumiya.render;

import com.google.common.base.Optional;
import com.google.common.base.Strings;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/* loaded from: classes.dex */
public class GPUDetection {
    public static final String GPU_FAMILY_ADRENO = "Adreno";
    public static final String GPU_FAMILY_TEGRA = "Tegra";
    public static final int INVALID_VERSION = -1;

    @Nonnull
    public final Optional<String> detectedFamily;
    public final int detectedNumericVersion;

    @Nonnull
    public final Optional<String> detectedVersion;

    public GPUDetection(@Nullable String str) {
        int i;
        String nullToEmpty = Strings.nullToEmpty(str);
        if (!nullToEmpty.toLowerCase().contains("adreno")) {
            if (nullToEmpty.toLowerCase().contains("tegra")) {
                this.detectedFamily = Optional.of(GPU_FAMILY_TEGRA);
                this.detectedVersion = Optional.absent();
                this.detectedNumericVersion = -1;
                return;
            } else {
                this.detectedFamily = Optional.absent();
                this.detectedVersion = Optional.absent();
                this.detectedNumericVersion = -1;
                return;
            }
        }
        this.detectedFamily = Optional.of(GPU_FAMILY_ADRENO);
        Matcher matcher = Pattern.compile(".*?Adreno.*?([0-9]+).*?", 2).matcher(nullToEmpty);
        if (!matcher.matches()) {
            this.detectedVersion = Optional.absent();
            this.detectedNumericVersion = -1;
        } else {
            this.detectedVersion = Optional.fromNullable(Strings.emptyToNull(matcher.group(1)));
            try {
                i = Integer.parseInt(this.detectedVersion.or((Optional<String>) ""));
            } catch (NumberFormatException e) {
                i = -1;
            }
            this.detectedNumericVersion = i;
        }
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/GPUDetection.kt
package com.lumiyaviewer.lumiya.render

import com.google.common.base.Optional
import com.google.common.base.Strings
import java.util.regex.Pattern

class GPUDetection(rendererString: String?) {
    @JvmField
    val detectedFamily: Optional<String>

    @JvmField
    val detectedNumericVersion: Int

    @JvmField
    val detectedVersion: Optional<String>

    init {
        val renderer = Strings.nullToEmpty(rendererString)
        if (renderer.lowercase().contains("adreno")) {
            detectedFamily = Optional.of(GPU_FAMILY_ADRENO)
            val matcher = Pattern.compile(".*?Adreno.*?([0-9]+).*?", Pattern.CASE_INSENSITIVE).matcher(renderer)
            if (matcher.matches()) {
                detectedVersion = Optional.fromNullable(Strings.emptyToNull(matcher.group(1)))
                detectedNumericVersion = try {
                    detectedVersion.or("").toInt()
                } catch (e: NumberFormatException) {
                    INVALID_VERSION
                }
            } else {
                detectedVersion = Optional.absent()
                detectedNumericVersion = INVALID_VERSION
            }
        } else if (renderer.lowercase().contains("tegra")) {
            detectedFamily = Optional.of(GPU_FAMILY_TEGRA)
            detectedVersion = Optional.absent()
            detectedNumericVersion = INVALID_VERSION
        } else {
            detectedFamily = Optional.absent()
            detectedVersion = Optional.absent()
            detectedNumericVersion = INVALID_VERSION
        }
    }

    companion object {
        const val GPU_FAMILY_ADRENO = "Adreno"
        const val GPU_FAMILY_TEGRA = "Tegra"
        const val INVALID_VERSION = -1
    }
}
